### DISCUSSION TITLE
Asset check and blocking auto-materialize

### DISCUSSION THREAD

**Discussion by ssillaots-boku** [dagster-io/dagster#22427] [Q&A]
I haven't found a similar topic yet but please point me to it if there's one.

Is there a way to make asset check failure block asset materialization for downstream asset if the downstream asset has auto-materialize?

Scenario:
Two assets: `upstream` and `downstream`. Upstream asset is a dbt asset which has a test. So the ideal flow would be like this:
1. upstream asset runs `dbt run`
2. upstream asset runs `dbt test`
3. dbt test fails
4. `downstream_` doesn't run

Right now 4th step doesn't work as expected and downstream asset starts materializing. 
You might say that why don't I add the same dbt test to `downstream` asset. Then the asset check would run before `downstream` asset rather than after `upstream` asset. It's totally fine to do that if you have 1 or maybe even 2 downstream assets. But the problem arises when you have a bunch (5+) of downstream assets that are dependent on the same upstream asset. You would then have to add the same test to all downstream assets instead of 1 test to the upstream asset. So a lot of duplication and tech debt for the future.

**Comment by ssillaots-boku** [comment]
Is this something Dagster will plan to implement in the future (blocking downstream asset's auto-materialize when upstream's asset check fails)?

**Reply by ssillaots-boku** [reply]
> Is this something Dagster will plan to implement in the future (blocking downstream asset's auto-materialize when upstream's asset check fails)?

Still hoping for an answer here 🤞

**Reply by OwenKephart** [reply]
Hi @ssillaots-boku ! Sorry for the late reply here, but yes this is something that we hope to accomplish with the AutomationCondition API: https://github.com/dagster-io/dagster/discussions/22811
